lib/astarte_data_access.ex

```diff
@@ -0,0 +1,44 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataAccess do
+  # Automatically defines child_spec/1
+  use Supervisor
+  alias Astarte.DataAccess.Config
+
+  def start_link(init_arg) do
+    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  @impl true
+  def init(init_arg) do
+    atom_keys? = Keyword.get(init_arg, :atom_keys, false)
```

Comment:
Thinking again about it, this is totally useless, since the whole library uses and assumes atoms throughout it so it just won't work if you actually pass `false` here.

Drop it and add a `TODO` where you set `:atom_keys` below